pages/src/rtl-switcher/rtl-switcher.ts

```diff
@@ -0,0 +1,81 @@
+import {ready} from '../../../src/modules/esl-utils/decorators/ready';
+import {attr} from '../../../src/modules/esl-base-element/decorators/attr';
+import {bind} from '../../../src/modules/esl-utils/decorators/bind';
+import {boolAttr, ESLBaseElement} from '../../../src/modules/esl-base-element/core';
+
+import {TraversingQuery} from '../../../src/modules/all';
+
+export class ESLDemoRTLSwitcher extends ESLBaseElement {
+  static is = 'esl-d-rtl-switcher';
+
+  protected _$target: any;
+
+  @boolAttr({readonly: true}) public active: boolean;
+
+  @attr({defaultValue: '::next'}) public target: string;
+
+  @ready
+  protected connectedCallback() {
+    super.connectedCallback();
+    this.updateTargetFromSelector();
+    this.setInitialDirection();
+  }
+
+  @ready
+  protected disconnectedCallback() {
+    super.disconnectedCallback();
+  }
+
+  public get $target(): any {
+    return this._$target;
+  }
+
+  public get $active(): any {
+    return this.$target?.dir === 'rtl';
+  }
+
+  public set $target(newPopupInstance) {
+    this._$target = newPopupInstance;
+    this.bindEvents();
```

Comment:
It's better to bing and unbind event in connected/disconnected callback and have a check for an existent target in event handler action